dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java

```diff
@@ -0,0 +1,163 @@
+            tasks = claimedTaskService.findByEperson(context, ep);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Page<ClaimedTaskRest> page = utils.getPage(tasks, pageable).map(converter);
+        return page;
+    }
+
+    @Override
+    public Class<ClaimedTaskRest> getDomainClass() {
+        return ClaimedTaskRest.class;
+    }
+
+    @Override
+    public ClaimedTaskResource wrapResource(ClaimedTaskRest task, String... rels) {
+        return new ClaimedTaskResource(task, utils, rels);
+    }
+
+    @Override
+    protected ClaimedTaskRest action(Context context, HttpServletRequest request, Integer id)
+        throws SQLException, IOException, AuthorizeException {
+        ClaimedTask task = null;
+        task = claimedTaskService.find(context, id);
+        XmlWorkflowServiceFactory factory = (XmlWorkflowServiceFactory) XmlWorkflowServiceFactory.getInstance();
+        Workflow workflow;
+        try {
+            workflow = factory.getWorkflowFactory().getWorkflow(task.getWorkflowItem().getCollection());
+
+            Step step = workflow.getStep(task.getStepID());
+            WorkflowActionConfig currentActionConfig = step.getActionConfig(task.getActionID());
+            workflowService
+                .doState(context, context.getCurrentUser(), request, task.getWorkflowItem().getID(), workflow,
+                    currentActionConfig);
+            if (!Action.getErrorFields(request).isEmpty()) {
+                throw new UnprocessableEntityException("Missing required fields");
+            }
+            // workflowRequirementsService.removeClaimedUser(context, task.getWorkflowItem(), task.getOwner(), task
+            // .getStepID());
+            context.addEvent(new Event(Event.MODIFY, Constants.ITEM, task.getWorkflowItem().getItem().getID(), null,
+                itemService.getIdentifiers(context, task.getWorkflowItem().getItem())));
+        } catch (WorkflowConfigurationException | MessagingException | WorkflowException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    @Override
+    protected void delete(Context context, Integer id) {
+        ClaimedTask task = null;
+        try {
+            task = claimedTaskService.find(context, id);
+            XmlWorkflowItem workflowItem = task.getWorkflowItem();
+            workflowService.deleteClaimedTask(context, workflowItem, task);
+            workflowRequirementsService.removeClaimedUser(context, workflowItem, task.getOwner(), task.getStepID());
+            context.addEvent(new Event(Event.MODIFY, Constants.ITEM, workflowItem.getItem().getID(), null,
+                itemService.getIdentifiers(context, workflowItem.getItem())));
+        } catch (SQLException | IOException | WorkflowConfigurationException | AuthorizeException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
We should return a more useful error here.  Something like `"Could not delete claimed task with ID" + id`.

Reply:
I have improved the error handling wrapping authorization exception in our equivalent REST exception. I disagree about add more details on the action here as the client already know what it was trying to do and this information can be added in the log in a central place to have a better uniformity (in our DSpaceApiExceptionControllerAdvice). Currently we never add this details wrapping the original exception so starting to do it now will complicate the future resolution of https://jira.duraspace.org/browse/DS-3992 (BTW I just noted that we have completely miss an old PR related to that)

Reply:
Withdrawing my request & resolving this conversation.